src/components/User/UserComponent.jsx

```diff
@@ -51,125 +52,86 @@ export default class UserComponent extends React.Component {
 
   retrieveUsers = (activePage, limit) => {
     if (checkIfCutoffExceeded(activePage, limit)) {
-      const url = `users?page=${activePage}&page_size=${limit}`;
-      this.props.loading(true);
-      return fetchData(url).then((response) => {
-        this.props.loading(false);
-        this.setState({ users: response.data.results });
-      }).catch(() => {
-        this.props.loading(false);
-        this.setState({ allDataFetched: true });
-      });
+      return this.makeAjaxRequest(activePage, limit);
     }
     return this.props.loadUsers(activePage, limit);
   };
 
-  handlePageTotal = () => Math.ceil(this.props.usersCount / this.state.limit);
+  makeAjaxRequest = async (activePage, limit) => {
+    const url = constructApiUrl('users', activePage, limit);
+    const response = await fetchInfo(url, this.props.loading);
+
+    const { results = {} } = response;
+    if (isEmpty(results)) {
+      this.setState({ allDataFetched: true });
+      return;
+    }
 
-  emptyUsersList = () => (isEmpty(this.props.users));
+    this.setState({ users: results });
+  }
```

Comment:
Just a nitpick. Could you terminate this statement? 😃

Reply:
🤣 sure